pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/MailboxReceiveNode.java

```diff
@@ -126,7 +133,8 @@ public boolean equals(Object o) {
       return false;
     }
     MailboxReceiveNode that = (MailboxReceiveNode) o;
-    return _senderStageId == that._senderStageId && _sort == that._sort && _sortedOnSender == that._sortedOnSender
+    return getSenderStageId() == that.getSenderStageId() && Objects.equals(_sender, that._sender)
```

Comment:
Looks like this change is causing all the test cases in `PlanNodeSerDeTest` to fail because the sender is not serialized into the plan.